tests/src/main/java/com/hazelcast/simulator/tests/synthetic/SyntheticTest.java

```diff
@@ -46,119 +45,115 @@
+ * pileup of back up commands which eventually can lead to an OOME.
  *
- * Another interesting scenario to test is a normal async invocation of a readonly operation (so no async/sync-backups) and see
- * if the system can be flooded with too many request. Normal sync operations don't cause that many problems because there is a
+ * Another interesting scenario to test is a normal async invocation of a readonly operation (so no async/sync-backups) and see if
+ * the system can be flooded with too many request. Normal sync operations don't cause that many problems because there is a
  * natural balance between the number of threads and the number of pending invocations.
  */
 public class SyntheticTest {
-    private static final ILogger log = Logger.getLogger(SyntheticTest.class);
 
-    //props
+    private static final ILogger LOGGER = Logger.getLogger(SyntheticTest.class);
+
+    // properties
     public boolean syncInvocation = true;
     public byte syncBackupCount = 0;
     public byte asyncBackupCount = 1;
     public long backupDelayNanos = 1000 * 1000;
     public boolean randomizeBackupDelay = true;
-    public int threadCount = 10;
     public int logFrequency = 10000;
     public int performanceUpdateFrequency = 1000;
     public KeyLocality keyLocality = KeyLocality.RANDOM;
+    public int keyCount = 1000;
     public int syncFrequency = 1;
     public String serviceName;
 
-    private AtomicLong operations = new AtomicLong();
-    private TestContext context;
     private HazelcastInstance targetInstance;
-    public IntervalProbe latency;
 
     @Setup
-    public void setup(TestContext context) throws Exception {
-        this.context = context;
-        this.targetInstance = context.getTargetInstance();
+    public void setup(TestContext testContext) throws Exception {
+        targetInstance = testContext.getTargetInstance();
     }
 
     @Teardown
     public void teardown() throws Exception {
-        log.info(getOperationCountInformation(targetInstance));
-        log.info(getPartitionDistributionInformation(targetInstance));
+        LOGGER.info(getOperationCountInformation(targetInstance));
+        LOGGER.info(getPartitionDistributionInformation(targetInstance));
     }
 
-    @Run
-    public void run() {
-        ThreadSpawner spawner = new ThreadSpawner(context.getTestId());
-        for (int k = 0; k < threadCount; k++) {
-            spawner.spawn(new Worker());
-        }
-        spawner.awaitCompletion();
+    @RunWithWorker
+    public Worker createWorker() {
+        return new Worker();
     }
 
-    @Performance
-    public long getOperationCount() {
-        return operations.get();
-    }
+    private class Worker implements IWorker, ExecutionCallback<Object> {
+
+        // these fields will be injected by the TestContainer
+        TestContext testContext;
+        IntervalProbe intervalProbe;
```

Comment:
Due to the `IWorker` interface we can benefit from the auto-injected `intervalProbe`, so we don't need the `AtomicInteger operations` to measure the performance anymore.